.github/workflows/ci.yml

```diff
@@ -1,62 +1,40 @@
 name: ci
+
 on:
   push:
     branches: [main]
-  pull_request:
-    branches: ['*']
 
 jobs:
   what-changed:
     runs-on: ubuntu-latest
     outputs:
       packages: ${{ steps.set-packages.outputs.packages }}
+    permissions:
+      contents: 'read'
+      actions: 'read'
+      pull-requests: 'read'
```

Comment:
`nrwl/nx-set-sha` needs these permissions to get the state of previous workflows/actions